src/types.rs

```diff
@@ -921,3 +921,87 @@ impl Debug for Log {
             .finish()
     }
 }
+
+/// An Ethereum Transaction Receipt
+#[derive(Clone, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TransactionReceipt {
+    #[serde(flatten)]
+    pub kind: TransactionReceiptKind,
+    /// The hash of the transaction that emitted this log.
+    pub transaction_hash: Digest,
+    /// The index of the transaction that emitted this log within the block.
+    pub transaction_index: U256,
+    /// The hash of the block containing the transaction.
+    pub block_hash: Digest,
+    /// The height of the block containing the transaction.
+    pub block_number: U256,
+    /// Address of transaction sender.
+    pub from: Address,
+    /// Transaction receipient (Null for contract creation).
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub to: Option<Address>,
+    /// The price paid post-execution by the transaction (i.e. base fee + priority fee).
+    /// Both fields in 1559-style transactions are *maximums* (max fee + max priority fee), the
+    /// amount that's actually paid by users can only be determined post-execution.
+    effective_gas_price: U256,
+    /// The sum of gas used by this transaction and all preceding transactions in the same block.
+    pub cumulative_gas_used: U256,
+    /// The amount of gas used for this specific transaction alone.
+    pub gas_used: U256,
+    /// Contract address created, or `None` if not a deployment.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub contract_address: Option<Address>,
+    /// Logs emitted by the transaction.
+    pub logs: Vec<Log>,
+    /// The log bloom filter.
+    pub logs_bloom: Bloom,
+    /// State root. Only present before activation of [EIP-658](https://eips.ethereum.org/EIPS/eip-658).
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub root: Option<Digest>,
+    /// Status: either 1 (success) or 0 (failure). Only present after activation of [EIP-658](https://eips.ethereum.org/EIPS/eip-658).
+    pub status: Option<U256>,
```

Comment:
This is another enum right? It's either one or the other:

```rust
enum Eip658 {
    Active {
        status: TransactionReceiptStatus,
    },
    BeforeActivation {
        root: Digest,
    },
}
```

**That being said**, since EIP-658 is active everywhere, do you think the distinction is worth it, or can we just assume that it is always a `status` and call it a day?

Also, for status here, just like for the `TransactionReceiptKind`, there are only two possible values: `0x1` and `0x0`, so we should model this as:

```rust
enum TransactionReceiptStatus {
    Failure,
    Success,
}
```

Reply:
I think it's so old that it's not worth separating. But we can do success and failure enum.

Reply:
Sounds good - I'm fine with getting rid of `root` field then, and then make `success` an enum instead of an integer.